app/common/auth/__init__.py

```diff
@@ -14,31 +16,52 @@
 )
 
 
-@auth_blueprint.route("/sign-in", methods=["GET", "POST"])
-def sign_in() -> ResponseReturnValue:
+@auth_blueprint.route("/request-a-link-to-sign-in", methods=["GET", "POST"])
+@auto_commit_after_request
+def request_a_link_to_sign_in() -> ResponseReturnValue:
     form = SignInForm()
     if form.validate_on_submit():
-        email = form.email_address.data
+        email = cast(str, form.email_address.data)
 
-        # TODO: all session stuff will be revisited as part of FSPT-334
-        session["email_address"] = email
+        user = get_or_create_user(email_address=email)
+
+        # TODO: read+validate redirect from query params
+        magic_link = interfaces.magic_link.create_magic_link(
+            user=user, redirect_to_path=url_for("platform.list_grants")
+        )
 
         notification_service.send_magic_link(
-            email,  # type: ignore[arg-type]
-            magic_link_url="https://magic-link-tbd",
-            magic_link_expires_at_utc=datetime.datetime.now(ZoneInfo("UTC")) + datetime.timedelta(minutes=15),
-            request_new_magic_link_url="https://new-magic-link-tbd",
+            email,
+            magic_link_url=url_for("auth.claim_magic_link", magic_link_code=magic_link.code, _external=True),
+            magic_link_expires_at_utc=magic_link.expires_at_utc,
+            request_new_magic_link_url=url_for("auth.request_a_link_to_sign_in", _external=True),
         )
 
-        return redirect(url_for("auth.check_email"))
+        return redirect(url_for("auth.check_email", magic_link_id=magic_link.id))
 
     return render_template("common/auth/sign_in.html", form=form)
 
 
-@auth_blueprint.get("/check-your-email")
-def check_email() -> ResponseReturnValue:
-    if "email_address" not in session:
-        return redirect(url_for("auth.sign_in"))
+@auth_blueprint.get("/check-your-email/<uuid:magic_link_id>")
+def check_email(magic_link_id: uuid.UUID) -> ResponseReturnValue:
+    magic_link = interfaces.magic_link.get_magic_link(id_=magic_link_id)
+    if not magic_link or not magic_link.usable:
```

Comment:
Link this to our convo on managing errors and default behaviour for things not being there/ having an incorrect ID etc. -- i.e in this case would it make sense for get magic link to only return usable valid links and otherwise not found is propagated and managed by the middleware (maybe not! having it explicit is fine too just another data point)

Reply:
Yeah interesting, didn't think of having `get_magic_link` also do a usable check. That might be v sensible you're right 👍 

I will consider this for a followup PR and hopefully not forget.

We will need to do a scan over all the interfaces soon-ish and think about error propagation as you say, so maybe we just do it in one pass later.